docs/tutorials/validate_data.rst

```diff
@@ -0,0 +1,447 @@
+
+                See the complete list of options for `Spark DataFrameReader <https://spark.apache.org/docs/latest/api/python/pyspark.sql.html#pyspark.sql.DataFrameReader>`__
+
+            .. tab-container:: tab2
+                :title: SQLAlchemy
+
+                A SQLAlchemy datasource can accept ``batch_kwargs`` that instruct it load a batch from a table, a view, or a result set of a query:
+
+                If you would like to validate an entire table (or a view) in your database's default schema:
+
+                .. code-block:: python
+
+                    batch_kwargs = {'table': "YOUR TABLE NAME"}
+
+                If you would like to validate an entire table or view from a non-default schema in your database:
+
+                .. code-block:: python
+
+                    batch_kwargs = {'table': "YOUR TABLE NAME", "schema": "YOUR SCHEMA"}
+
+                If you would like to validate using a query to construct a temporary table:
+
+                .. code-block:: python
+
+                    batch_kwargs = {'query': 'SELECT YOUR_ROWS FROM YOUR_TABLE'}
+
+
+        The examples of ``batch_kwargs`` above can also be the outputs of "generators" used by Great Expectations. You can read about the default Generators' behavior and how to implement additional generators in this article: :ref:`batch_generator`.
+
+
+|
+Now you have the contents of one of the files loaded as batch of the data asset ``data__dir/default/npidata``.
+
+
+5. Set a Run Id
+---------------
+
+A ``run_id`` links together validations of different data assets, making it possible to track "runs" of a pipeline and
+follow data assets as they are transformed, joined, annotated, enriched, or evaluated. The run id can be any string;
+by default, Great Expectations will use an ISO 8601-formatted UTC datetime string.
+
+
+The default ``run_id`` generated by Great Expectations is built using the following code:
+
+.. code-block:: python
+
+    run_id = datetime.datetime.utcnow().isoformat().replace(":", "") + "Z"
+
+
+When you integrate validation in your pipeline, your pipeline runner probably has a run id that can be inserted here to make smoother integration.
+
+6. Validate the batch
+---------------------
+
+Validation evaluates our expectations against the given batch and produces a report that describes observed values and
+any places where expectations are not met. To validate the batch of data call the :meth:`~great_expectations.\
+data_asset.data_asset.DataAsset.validate` method on the batch:
+
+.. code-block:: python
+
+  validation_result = batch.validate(run_id=run_id)
+
+
+In a data pipeline we may take specific actions based on the the result of the validation.
+
+A common pattern is to check ``validation_result``'s ``success`` key (``True`` if the batch meets all the expectations in the expectation suite), and stop or issue a warning in the code in case of failure:
```

Comment:
```suggestion
A common pattern is to check the ``validation_result``'s ``success`` key (``True`` if the batch meets all the expectations in the expectation suite), and stop or issue a warning in the code in case of failure:
```